plugins/uefi/fu-uefi-tool.c

```diff
@@ -0,0 +1,312 @@
+	gboolean action_supported = FALSE;
+	gboolean action_unset_debug = FALSE;
+	gboolean action_version = FALSE;
+	gboolean force = FALSE;
+	gboolean ret;
+	gboolean verbose = FALSE;
+	g_autofree gchar *apply = FALSE;
+	g_autofree gchar *esp_path = NULL;
+	g_autoptr(FuUtilPrivate) priv = g_new0 (FuUtilPrivate, 1);
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+	const GOptionEntry options[] = {
+		{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
+			/* TRANSLATORS: command line option */
+			_("Show extra debugging information"), NULL },
+		{ "version", '\0', 0, G_OPTION_ARG_NONE, &action_version,
+			/* TRANSLATORS: command line option */
+			_("Display action_version"), NULL },
+		{ "force", 'f', 0, G_OPTION_ARG_NONE, &force,
+			/* TRANSLATORS: command line option */
+			_("Forces flash even if GUID is not in the ESRT"), NULL },
+		{ "log", 'L', 0, G_OPTION_ARG_NONE, &action_log,
+			/* TRANSLATORS: command line option */
+			_("Forces flash even if GUID is not in the ESRT"), NULL },
+		{ "list", 'l', 0, G_OPTION_ARG_NONE, &action_list,
+			/* TRANSLATORS: command line option */
+			_("List action_supported firmware updates"), NULL },
+		{ "supported", 's', 0, G_OPTION_ARG_NONE, &action_supported,
+			/* TRANSLATORS: command line option */
+			_("Query for firmware update support"), NULL },
+		{ "info", 'i', 0, G_OPTION_ARG_NONE, &action_info,
+			/* TRANSLATORS: command line option */
+			_("Show the information of firmware update status"), NULL },
+		{ "enable", 'e', 0, G_OPTION_ARG_NONE, &action_enable,
+			/* TRANSLATORS: command line option */
+			_("Enable firmware update support on action_supported systems"), NULL },
+		{ "esp-path", 'e', 0, G_OPTION_ARG_STRING, &esp_path,
+			/* TRANSLATORS: command line option */
+			_("Override the default ESP path"), "PATH" },
+		{ "set-debug", 'd', 0, G_OPTION_ARG_NONE, &action_set_debug,
+			/* TRANSLATORS: command line option */
+			_("Set the debugging flag during update"), NULL },
+		{ "unset-debug", 'D', 0, G_OPTION_ARG_NONE, &action_unset_debug,
+			/* TRANSLATORS: command line option */
+			_("Unset the debugging flag during update"), NULL },
+		{ "apply", 'a', 0, G_OPTION_ARG_STRING, &apply,
+			/* TRANSLATORS: command line option */
+			_("Apply firmware updates"), "GUID" },
+		{ NULL}
+	};
+
+	setlocale (LC_ALL, "");
+
+	bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+	textdomain (GETTEXT_PACKAGE);
+
+	/* ensure root user */
+	if (getuid () != 0 || geteuid () != 0)
+		/* TRANSLATORS: we're poking around as a power user */
+		g_printerr ("%s\n", _("This program may only work correctly as root"));
+
+	/* get a action_list of the commands */
+	priv->context = g_option_context_new (NULL);
+	g_option_context_set_description (priv->context,
+		"This tool allows an administrator to debug UpdateCapsule operation.");
+
+	/* TRANSLATORS: program name */
+	g_set_application_name (_("Firmware Utility"));
```

Comment:
It should probably be clearer that this is `UEFI firmware utility` since a layman finding it in libexec might not know what it is.